src/meta/processors/admin/AdminClient.cpp

```diff
@@ -335,6 +335,77 @@ StatusOr<std::vector<HostAddr>> AdminClient::getPeers(GraphSpaceID spaceId, Part
     return Status::Error("Get Failed");
 }
 
+folly::Future<Status> AdminClient::getLeaderDist(HostLeaderMap* result) {
+    if (injector_) {
+        return injector_->getLeaderDist(result);
+    }
+    folly::Promise<Status> promise;
+    auto future = promise.getFuture();
+
+    const auto& prefix = MetaServiceUtils::hostPrefix();
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto ret = kv_->prefix(kDefaultSpaceId, kDefaultPartId, prefix, &iter);
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
+        promise.setValue(Status::Error("Can't find any hosts"));
+        return future;
+    }
+
+    std::vector<HostAddr> allHosts;
+    while (iter->valid()) {
+        auto key = iter->key();
+        IPv4 ip;
+        Port port;
+        memcpy(&ip, key.data() + prefix.size(), sizeof(IPv4));
+        memcpy(&port, key.data() + prefix.size() + sizeof(IPv4), sizeof(Port));
+        iter->next();
+        allHosts.emplace_back(ip, port);
+        (*result)[allHosts.back()] = {};
+    }
+
+    std::vector<folly::SemiFuture<Status>> hostFutures;
+    for (const auto& h : allHosts) {
+        auto getLeader = [result, this] (const HostAddr& host) {
+            storage::cpp2::GetLeaderReq req;
+            folly::Promise<Status> pro;
+            auto f = pro.getFuture();
+            auto* evb = ioThreadPool_->getEventBase();
+            auto client = clientsMan_->client(host, evb);
+            client->future_getLeaderPart(std::move(req))
```

Comment:
overload getResponse?

Reply:
For now response type is not same, I'll modify it later.